File: crates/aegis-language/tests/corpora/python/exec_shell.py
# exec_shell.py — positive corpus: process / shell execution sinks.
#
# Expected: six CodeExecution operations. Each first positional argument is a
# string literal → Known certainty, and the literal is a shell command → a
# nested recursive target in Bash (ADR-022 §7 cross-language nesting). Covers
# os.system and the subprocess.{run,call,Popen,check_call,check_output} family.
# Parses cleanly.
import os
import subprocess

os.system("rm -rf /tmp/x")
subprocess.run("rm /tmp/y")
subprocess.call("rm /tmp/z")
subprocess.Popen("rm /tmp/w")
subprocess.check_call("rm /tmp/v")
subprocess.check_output("rm /tmp/u")

File: crates/aegis-language/tests/corpora/typescript/exec_shell.ts
// Corpus: execution sinks producing Bash (shell) payloads (plan Iteration 7,
// ADR-022 §3, §7 cross-language nesting).
//
// `child_process.exec` and `execSync` run a literal argument through a shell,
// so a string-literal argument is recursively analyzable as Bash. Each fires
// as `CodeExecution` with `Known` certainty and a Bash nested payload carrying
// the literal source (escapes as-written; no decoding this slice).
//
// TypeScript enrichment: the third call carries an explicit type argument
// (`child_process.exec<void>(...)`). The query still binds `function` to the
// `member_expression` and the first positional argument is still the payload,
// so the Bash nested target is recovered unchanged.

child_process.exec("rm -rf /tmp/x");
child_process.execSync("rm /tmp/y");
child_process.exec<void>("rm -rf /tmp/z");